src/cascadia/TerminalSettingsEditor/ViewModelHelpers.h

```diff
@@ -39,24 +39,28 @@ struct ViewModelHelper
     winrt::event<::winrt::Windows::UI::Xaml::Data::PropertyChangedEventHandler> _propertyChangedHandlers;
 };
 
-#define _BASE_OBSERVABLE_PROJECTED_SETTING(target, name)  \
-public:                                                   \
-    auto name() const noexcept { return target.name(); }; \
-    template<typename T>                                  \
-    void name(const T& value)                             \
-    {                                                     \
-        if (name() != value)                              \
-        {                                                 \
-            target.name(value);                           \
-            _NotifyChanges(L"Has" #name, L## #name);      \
-        }                                                 \
-    }                                                     \
+#define _GET_SETTING_FUNC(target, name) \
```

Comment:
A bit annoyed by this change. I wanted to use `OBSERVABLE_PROJECTED_SETTING` for Default Profile (I think?), but the `ClearXXX` function wasn't valid. So I had to break things up a bit here.

Reply:
wait isn't that why we have the `PERMANENT` variant of `OBSERVABLE...`? Like, Name and Guid cannot be cleared.

Reply:
yea this is probably the only one I'm not signing off over. Not blocking, over, but it is a little weird

Reply:
Sorry, the exact problem is that `DefaultProfile` only has a getter and a setter (because we do that dance where we get the unparsed default profile, but we _always_ have a default profile). So it doesn't use inheritance at all.

The `PERMANENT` variant gets rid of `ClearXXX`, but still uses the `HasXXX` API. I could introduce a third macro called `GET_SET_XXX` that would just expose the getter/setter. Idk how valuable that would be though tbh.